src/qibolab/instruments/zhinst.py

```diff
@@ -0,0 +1,1050 @@
+
+    # TODO: Play taking a big sequence with several acquire steps
+    def play(self, qubits, sequence, options):
+        """Play pulse sequence"""
+        self.signal_map = {}
+        dimensions = []
+        if options.averaging_mode is AveragingMode.SINGLESHOT:
+            dimensions = [options.nshots]
+
+        # TODO: Read frequency for pulses instead of qubit patch
+        for qubit in qubits.values():
+            for pulse in sequence:
+                if pulse.qubit == qubit.name:
+                    if pulse.type is PulseType.READOUT:
+                        qubit.readout_frequency = pulse.frequency
+                    if pulse.type is PulseType.DRIVE:
+                        qubit.drive_frequency = pulse.frequency
+
+        """
+        Play pulse sequence steps, one on each method:
+        Translation, Calibration, Experiment Definition and Execution.
+        """
+        self.sequence_zh(sequence, qubits, sweepers=[])
+        self.calibration_step(qubits)
+        self.create_exp(qubits, options)
+        self.run_exp()
+
+        # TODO: General, several readouts and qubits
+        # TODO: Implement the new results!
+        "Get the results back"
+        results = {}
+        for qubit in qubits.values():
+            if not qubit.flux_coupler:
+                if self.sequence[f"readout{qubit.name}"]:
+                    exp_res = self.results.get_data(f"sequence{qubit.name}")
+                    if options.acquisition_type is AcquisitionType.DISCRIMINATION:
+                        if options.averaging_mode is AveragingMode.CYCLIC:
+                            data = np.array([exp_res])
+                        else:
+                            data = np.array(exp_res)
+                        results[self.sequence[f"readout{qubit.name}"][0].pulse.serial] = options.results_type(data)
+                        results[self.sequence[f"readout{qubit.name}"][0].pulse.qubit] = options.results_type(data)
+                    else:
+                        results[self.sequence[f"readout{qubit.name}"][0].pulse.serial] = options.results_type(
+                            data=np.array(exp_res)
+                        )
+                        results[self.sequence[f"readout{qubit.name}"][0].pulse.qubit] = options.results_type(
+                            data=np.array(exp_res)
+                        )
+
+        exp_dimensions = list(np.array(exp_res).shape)
+        if dimensions != exp_dimensions:
+            print("dimensions", dimensions, "experiment", exp_dimensions)
+            warnings.warn("dimensions not properly ordered")
+
+        # FIXME: Include this on the reports
+        # html containing the pulse sequence schedule
+        # lo.show_pulse_sheet("pulses", self.exp)
+
+        # There is no reason for disconnection and it prevents reconnection
+        # for a period of time making the software loops with execute_play_sequence crash
+        # self.disconnect()
```

Comment:
Dead code

Reply:
the pulse sheet I will leave it as it can be really useful when debugging new routines in qibocal. At least I found it that way but should not be enabled by default as to avoid creating useless files.